src/tds/data.c

```diff
@@ -723,6 +722,22 @@ tds_generic_get(TDSSOCKET * tds, TDSCOLUMN * curcol)
 	tdsdump_log(TDS_DBG_INFO1, "tds_get_data: type %d, varint size %d\n", curcol->column_type, curcol->column_varint_size);
 	switch (curcol->column_varint_size) {
 	case 4:
+		if (!is_blob_type(curcol->column_type)) {
+			/* Any other non-BLOB type (e.g., XSYBCHAR) */
```

Comment:
Varint for `XSYBCHAR` is 2, not 4.
Is not this if handling also below case (`SYBLONGBINARY`) ?
To be honest I would invert varint `4` and `5`, meaning "old" blobs (`SYBTEXT`, `SYBNTEXT`, `SYBIMAGE`) having a length for pointer (1 byte) and one for data (4 bytes) marked with 5 (1+4) and "new" blobs (`SYBLONGBINARY` and similar) with 4.

Reply:
Oops, yeah, in retrospect I should have called out `SYBLONG{BINARY,CHAR}` here, and the `SYBLONGBINARY` code was indeed unused.